File: db/src/main/java/com/delivery/db/storemenu/StoreMenuRepository.java
package com.delivery.db.storemenu;

import com.delivery.db.storemenu.enums.StoreMenuStatus;
import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.support.JpaRepositoryImplementation;

public interface StoreMenuRepository extends JpaRepositoryImplementation<StoreMenuEntity, Long> {

    Optional<StoreMenuEntity> findFirstByIdAndStatusOrderByIdDesc(Long id, StoreMenuStatus status);

    List<StoreMenuEntity> findAllByStoreIdAndStatusOrderBySequenceDesc(Long storeId,
        StoreMenuStatus status);

}

File: db/src/main/kotlin/com/delivery/db/storemenu/StoreMenuRepository.kt
package com.delivery.db.storemenu

import com.delivery.db.storemenu.enums.StoreMenuStatus
import org.springframework.data.jpa.repository.support.JpaRepositoryImplementation
import java.util.*

interface StoreMenuRepository : JpaRepositoryImplementation<StoreMenuEntity, Long> {
    fun findFirstByIdAndStatusOrderByIdDesc(
        id: Long?,
        status: StoreMenuStatus?
    ): StoreMenuEntity?

    fun findAllByStoreIdAndStatusOrderBySequenceDesc(
        storeId: Long?,
        status: StoreMenuStatus?
    ): List<StoreMenuEntity>
}
